app/experimenter/settings.py

```diff
@@ -9,12 +9,26 @@
 For the full list of settings and their values, see
 https://docs.djangoproject.com/en/1.9/ref/settings/
 """
+from django.contrib.admin import ModelAdmin, StackedInline, TabularInline
+from django.db.models import DecimalField, ForeignKey, JSONField, ManyToManyField
+
+for cls in [
+    DecimalField,
+    ForeignKey,
+    JSONField,
+    ManyToManyField,
+    ModelAdmin,
+    StackedInline,
+    TabularInline,
+]:
+    cls.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)
+
+
```

Comment:
What is this for?

Reply:
So this is an unfortunate part of where typechecking in Python is at right now.  So the theory is adding static type checking in Python is, the type annotations should be valid Python code, but also basically do nothing at runtime, ie be ignored.  Usually that's things like

```py
x: int = 1
```

or

```py
def my_func() -> int:
    return 1
```

The type annotations there `: int` or `-> int` can be ignored at runtime.  But another syntax that Python static typing uses is to add a generic parameter to something which looks like

```py
things: Dict[str, int] = {}
```

Notice the square brackets there.  The type checker is using those as a way to say 'these are the generic type specifications for this type', but square brackets also have runtime meaning in Python, which is it triggers a `__get_item__` method call, which is what's happening when you say `some_dictionary[some_key]`.  So what happens is at runtime, it tries to actually call `__get_item__` even though you're not trying to actually do like a dictionary lookup, you're just specifying generic types.  

So what this means is

```py
    cls.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)
```

means 'If someone adds a generic type parameter to these classes using [...], just treat it as a no op and do nothing at runtime'

It's kind of an annoying hack that's a side effect of the design of Python static type checking.

Reply:
Interesting 🤔

Reply:
Great explanation, ill need to save that one!